packages/arcgis-rest-request/src/job.ts

```diff
@@ -0,0 +1,494 @@
+   */
+  get pollingRate() {
+    return this._pollingRate;
+  }
+
+  /**
+   * Setter that sets a new polling rate if the user changes the rate during the job task.
+   */
+  set pollingRate(newRate: number) {
+    this.stopEventMonitoring();
+    this.startEventMonitoring(newRate);
+  }
+
+  /**
+   * Retrieves the status of the current job. 
+   * 
+   * @returns An object with the job id and jobStatus.
+   */
+  getJobInfo() {
+    return request(this.jobUrl, {
+      authentication: this.authentication
+    });
+  }
+
+  /**
+   * Function that calls the {@linkcode Job.getJobInfo} to check the job status, and emits the current job status. There are custom event emitters that
+   * the user is able to listen based on the job status. Refer to {@linkcode JOB_STATUSES} to see the various enums of the job status.
+   * To get results array from the job task, the job status must be {@linkcode JOB_STATUSES.Success}.
+   * 
+   * These job statuses are based on what are returned from the job request task and have been into an enum type in {@linkcode JOB_STATUSES}.
+   * 
+   * Reference {@link https://developers.arcgis.com/rest/services-reference/enterprise/checking-job-status.html}
+   * 
+   * @returns An object with the job id, job status, and messages array with a status property of type and description.
+   */
+  private executePoll = async () => {
+    let result;
+    try {
+      result = await this.getJobInfo();
+    } catch (error) {
+      this.emitter.emit(JOB_STATUSES.Error, error);
+      return;
+    }
+    this.emitter.emit(JOB_STATUSES.Status, result);
+
+    switch (result.jobStatus) {
+      case "esriJobCancelled":
+        this.emitter.emit(JOB_STATUSES.Cancelled, result);
+        break;
+      case "esriJobCancelling":
+        this.emitter.emit(JOB_STATUSES.Cancelling, result);
+        break;
+      case "esriJobNew":
+        this.emitter.emit(JOB_STATUSES.New, result);
+        break;
+      case "esriJobWaiting":
+        this.emitter.emit(JOB_STATUSES.Waiting, result);
+        break;
+      case "esriJobExecuting":
+        this.emitter.emit(JOB_STATUSES.Executing, result);
+        break;
+      case "esriJobSubmitted":
+        this.emitter.emit(JOB_STATUSES.Submitted, result);
+        break;
+      case "esriJobTimedOut":
+        this.emitter.emit(JOB_STATUSES.TimedOut, result);
+        break;
+      case "esriJobFailed":
+        this.emitter.emit(JOB_STATUSES.Failed, result);
+        break;
+      case "expectedFailure":
+        this.emitter.emit(JOB_STATUSES.Failure, result);
+        break;
+      case "esriJobSucceeded":
+        this.emitter.emit(JOB_STATUSES.Success, result);
+        break;
+    }
+  };
+
+  /**
+   * A handler that listens for an eventName and returns custom handler.
+   * 
+   * @param eventName A string of what event to listen for.
+   * @param handler A function of what to do when eventName was called.
+   */
+  on(eventName: string, handler: (e: any) => void) {
```

Comment:
Once you build the `IJobInfo` interface we can use it here instead of `any`

```suggestion
  on(eventName: string, handler: (e: IJobInfo) => void) {
```